tests/CMakeLists.txt

```diff
@@ -169,18 +169,28 @@ FOREACH(_test ${_tests})
       SET(_testlib)
     ENDIF()
 
-    # Create the output directory and a rule to generate the input file therein.
-    # This input file is the one found in the source dir, but it has the
-    # correct output directory so that we do not need to specify it by hand
-    # (a common mistake when copying a previous test) and do not need to
-    # specify any additional plugin shared libraries, should they exist
-    # for this test
+    # Create the output directory and subdirectories if necessary
+    FILE(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/output-${_test})
+
+    FILE(GLOB_RECURSE _outputs RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}/${_test}/
+	${CMAKE_CURRENT_SOURCE_DIR}/${_test}/[a-zA-Z0-9]*)
+    FOREACH(_output ${_outputs})
+      GET_FILENAME_COMPONENT(_dir ${_output} DIRECTORY)
+      IF(NOT  ${_dir} STREQUAL "" )
```

Comment:
Hm, is this the canonical way to figure out whether `${_dir}` is a directory? It's also not quite clear to me what would happen if we had directories that are nested deeper. I think this only works because you get the `_outputs` with the `RELATIVE` flag, and so the only way the `DIRECTORY` part of `_dir` is empty if you have a regular file in the current directory. But what if you had a *directory* in a subdirectory? Or a regular file in a subdirectory?